scripts/check_types.py

```diff
@@ -0,0 +1,104 @@
+"""
+Type hints checker using mypy.
+"""
+
+import json
+import logging
+import subprocess
+import sys
+import tempfile
+from collections import defaultdict
+from pathlib import Path
+
+# Set up logging
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+
+def main():
+    # Setup
+    script_dir = Path(__file__).resolve().parent
+    dirs = sys.argv[1:] or [
```

Comment:
These aren't necessarily going to be directories so should be named `paths` instead for clarity